### PR TITLE
app-text/mupdf: bump to 1.10a

### PR DESCRIPTION
**Pull request by charIes17**
app-text/mupdf: bump to version 1.10a

This fixes CVE-2016-6265, see https://bugs.gentoo.org/show_bug.cgi?id=589826
Also bumps to EAPI=6

Successful build log: https://bpaste.net/show/28a41124ad57

Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=589826  (CVE-2016-6265) 
Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=600674
Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=600928

### CODE REVIEW

**Comment by charIes17** [comment]
Some of those 9 patches from version 1.9a don't work. I have already reworked two of them.
But still have problems ( do not really understand what it does ) with **no-thirdlibs.patch**.  
Need help with that patch.  Also for those two conditional patches within src_prepare.

**Comment by charIes17** [comment]
Lots of older patches have been adjusted so they all get applied to version 1.10a.  
Now there come errors in compile phase.  Need advice.

**Comment by charIes17** [comment]
Learned some things about using git, Makefile, eclasses and patches.
Now compilation works,but only with bundled openjpeg.  With Gentoo installed openjpeg, it still throws the aforementioned errors.
Could use some help with clean up of the commented out bottom part of src_prepare.

**Comment by charIes17** [comment]
Still "work in progress".  Some more clean up needed.

**Comment by gentoolinux** [comment]
Thank you!

Could you please make the dev-lang/mujs dependency optional (add javascript USE flag)? It should be possible to disable JavaScript at compile time with FZ_ENABLE_JS=0.

**Comment by charIes17** [comment]
With reference to https://bugs.gentoo.org/show_bug.cgi?id=585062 I doubt that would work.
See https://gitweb.gentoo.org/repo/gentoo.git/diff/app-text/mupdf/mupdf-1.9a.ebuild?id=685bce928cda0b6063cc4ba3c38686472eb128b6

**Comment by charIes17** [comment]
Just tried with

    use javascript || \
        sed -e '/* #define FZ_ENABLE_JS/ adefine FZ_ENABLE_JS=0' \
            -i include/mupdf/fitz/config.h


Result is not very promising, see https://bpaste.net/show/21811491606f

**Comment by gentoolinux** [comment]
I guess you have to use
```
use javascript || \
    sed -e '/* #define FZ_ENABLE_JS/ a#define FZ_ENABLE_JS 0'
        -i include/mupdf/fitz/config.h
```
However, your ebuild does not work for me. The file `debian/mupdf.pc` is generated inside the main directory and not in `platform/`. Therefore applying the patch `mupdf-1.3-pkg-config.patch` fails.
Unfortunately I don't have time to investigate this further.

**Comment by charIes17** [comment]
What about the two other files generated from mupdf-1.5-old-debian-files.patch?
There should be:
./platform/debian/mupdf.desktop
./platform/debian/mupdf.pc
./platform/debian/mupdf.xpm

I had to add leading ./ to them because PATCHES array failed without.

Regarding the javascript matter I prefer to leave it as it is in version 1.9a,
See https://bugs.gentoo.org/show_bug.cgi?id=585062

**Comment by gentoolinux** [comment]
Thanks for the hint. I accidentally used the file `mupdf-1.5-old-debian-files.patch` from the portage tree and not yours. Sorry, your ebuild works fine.

Regarding javascript: Removing the javascript USE flag for version 1.9a was correct, but starting with version 1.10 it is now possible to build MuPDF without javascript (see https://www.mupdf.com/news). I have tested it and MuPDF compiles and works fine without mujs using the modified ebuild https://github.com/gentoolinux/gentoo/commit/cc3e357e3900324f931860553d5b07d4fd791fb8. So I propose to merge this ebuild. However, someone should review it before merging.

**Comment by charIes17** [comment]
Thanks for you contribution.  I have tested you proposal with USE=javascript and USE=-javascript with and without mujs installed. And I confirm it works for me.  So I integrated your proposal.

File: app-text/mupdf/mupdf-1.10a.ebuild [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	media-libs/openjpeg:2[static-libs?]
+	net-misc/curl[static-libs?]
+	virtual/jpeg[static-libs?]
+	X? ( x11-libs/libX11[static-libs?]
+		x11-libs/libXext[static-libs?] )
+	opengl? ( >=media-libs/glfw-3 )"
+RDEPEND="${LIB_DEPEND}"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+	static-libs? ( ${LIB_DEPEND} )
```
Comment (nvinson) [inline app-text/mupdf/mupdf-1.10a.ebuild]: Is the above line necessary?  It looks redundant to me.

File: app-text/mupdf/mupdf-1.10a.ebuild [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	my_soname=libmupdf.so.${PV}
+	my_soname_js_none=libmupdf-js-none.so.${PV}
+	sed -e "\$a\$(MUPDF_LIB): \$(MUPDF_JS_NONE_LIB)" \
+		-e "\$a\\\t\$(QUIET_LINK) \$(CC) \$(LDFLAGS) --shared -Wl,-soname -Wl,${my_soname} -Wl,--no-undefined -o \$@ \$^ \$(MUPDF_JS_NONE_LIB) \$(LIBS)" \
+		-e "/^MUPDF_LIB =/s:=.*:= \$(OUT)/${my_soname}:" \
+		-e "\$a\$(MUPDF_JS_NONE_LIB):" \
+		-e "\$a\\\t\$(QUIET_LINK) \$(CC) \$(LDFLAGS) --shared -Wl,-soname -Wl,${my_soname_js_none} -Wl,--no-undefined -o \$@ \$^ \$(LIBS)" \
+		-e "/install/s: COPYING : :" \
+		-i Makefile || die
+}
```
Comment (nvinson) [inline app-text/mupdf/mupdf-1.10a.ebuild]: It's not required, but I'd like to see all these sed replacements eventually converted into proper patches and submitted upstream.

Comment (charIes17) [inline app-text/mupdf/mupdf-1.10a.ebuild]: This is just a version bump, nothing else.  Feel free to send patches to upstream.

**Comment by xmw** [comment]
commit 290927105365ff1f2374f383d7135ecf17f41cb1
Author: Michael Weber <xmw@gentoo.org>
Date:   Mon Jan 23 01:31:02 2017 +0100

    app-text/mupdf: Version bump (https://github.com/gentoo/gentoo/pull/3108, thanks charIes17).
    
    Package-Manager: Portage-2.3.3, Repoman-2.3.1

**Comment by charIes17** [comment]
Sorry for not being more responsive to your proposal.  My intention was to get the version bump into the main repository as fast as possible.  And do more cleanup work later with another (or some more) pull request.  

Now as it is pulled in, we could do the clean-up.  What I'd like to to is removing all that my_soname stuff and use a conditional switch for the $(OUT) in Makefile instead.  Linuxfromscratch ( http://www.linuxfromscratch.org/blfs/view/cvs/pst/mupdf.html ) recommends a patch ( http://www.linuxfromscratch.org/patches/blfs/svn/mupdf-1.10a-shared_libs-1.patch ) that could possibly be used.  

How do you think about?

**Pull request closed without merging**